Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp

```diff
@@ -752,13 +751,15 @@ static void webkit_settings_class_init(WebKitSettingsClass* klass)
      * WebKitSettings:enable-java:
      *
      * Determines whether or not Java is enabled on the page.
+     *
+     * Deprecated: 2.38
      */
     sObjProperties[PROP_ENABLE_JAVA] =
         g_param_spec_boolean(
             "enable-java",
             _("Enable Java"),
             _("Whether Java support should be enabled."),
-            TRUE,
+            FALSE,
             readWriteConstructParamFlags);
```

Comment:
Oh, we forgot something here:

```suggestion
            static_cast<GParamFlags>(readWriteConstructParamFlags | G_PARAM_DEPRECATED));
```

Sadly, it's missing for a bunch of properties: enable-accelerated-2d-canvas, enable-plugins, and enable-xss-auditor. I will fix those separately, so you only need to worry about enable-java here.

Reply:
https://bugs.webkit.org/show_bug.cgi?id=240993